Otherwise it fails with older Yocto with following error (doesn't fail with 3.2 Gatesgarth):

| /jenkins/home/workspace/jansa/ros2-dashing-thud/ros2-dashing-thud/tmp-glibc/work/aarch64-oe-linux/cartographer/1.0.0-1-r0/recipe-sysroot/usr/include/eigen3/Eigen/src/Core/DenseStorage.h: In member function 'virtual void cartographer::sensor::{anonymous}::Collator_Ordering_Test::TestBody()':
| /jenkins/home/workspace/jansa/ros2-dashing-thud/ros2-dashing-thud/tmp-glibc/work/aarch64-oe-linux/cartographer/1.0.0-1-r0/recipe-sysroot/usr/include/eigen3/Eigen/src/Core/DenseStorage.h: In member function 'virtual void cartographer::sensor::{anonymous}::Collator_OrderingMultipleTrajectories_Test::TestBody()':
| /jenkins/home/workspace/jansa/ros2-dashing-thud/ros2-dashing-thud/tmp-glibc/work/aarch64-oe-linux/cartographer/1.0.0-1-r0/recipe-sysroot/usr/include/eigen3/Eigen/src/Core/DenseStorage.h:194:66: error: '*((void*)(&<anonymous>)+8).Eigen::DenseStorage<double, 3, 3, 1, 0>::m_data' may be used uninitialized in this function [-Werror=maybe-uninitialized]
|      DenseStorage(const DenseStorage& other) : m_data(other.m_data) {
|                                                                   ^
....
| cc1plus: some warnings being treated as errors

Upstream-Status: Pending

diff --git a/cmake/functions.cmake b/cmake/functions.cmake
index cdbd896..fa265ca 100644
--- a/cmake/functions.cmake
+++ b/cmake/functions.cmake
@@ -91,7 +91,6 @@ macro(google_initialize_cartographer_project)
     google_add_flag(GOOG_CXX_FLAGS "-Werror=reorder")
     google_add_flag(GOOG_CXX_FLAGS "-Werror=return-type")
     google_add_flag(GOOG_CXX_FLAGS "-Werror=switch")
-    google_add_flag(GOOG_CXX_FLAGS "-Werror=uninitialized")
 
     if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
       google_add_flag(GOOG_CXX_FLAGS "-Wthread-safety")
